Code/Tools/ProjectManager/Source/ProjectExportController.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <ProjectInfo.h>
+
+#include <QThread>
+#endif
+
+QT_FORWARD_DECLARE_CLASS(QProcess)
+
+namespace O3DE::ProjectManager
+{
+    QT_FORWARD_DECLARE_CLASS(ProjectButton)
+    QT_FORWARD_DECLARE_CLASS(ProjectExportWorker)
+
+    class ProjectExportController : public QObject
+    {
+        Q_OBJECT
+
+    public:
+        explicit ProjectExportController(const ProjectInfo& projectInfo, ProjectButton* projectButton, QWidget* parent = nullptr);
+        ~ProjectExportController();
+
+        void SetProjectButton(ProjectButton* projectButton);
+        const ProjectInfo& GetProjectInfo() const;
+
+        inline constexpr static int s_maxDisplayedBuiltOutputChars = 25;
+        inline static const char * LauncherExportFailedMessage = "Launcher failed to export.";
+
+    public slots:
+        void Start();
+        void UpdateUIProgress(const QString& lastLine);
+        void HandleResults(const QString& result);
+        void HandleCancel();
+
+    signals:
+        void Done(bool success = true);
+        void NotifyExportProject(const ProjectInfo& projectInfo);
+
+    private:
+        ProjectInfo m_projectInfo;
```

Comment:
you can probably declare this as a const reference

Reply:
Turns out I cannot. One of the members, `m_logUrl` needs to be updated whenever the worker finishes the task and updates logs.

Reply:
But if thats the case, it is updating the copy's `m_logUrl`, not the original projectInfo's copy because its not a reference. I don't know what affect that will have.